### PR TITLE
Linter

### PR DESCRIPTION
**Pull request by obelic2**
Added ESLint to repository as per Issue #29

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,31 +21,39 @@
...
-    "setup": "npm install -g mocha npx && npm install copyfiles -g && npm install -g delete && npm install -g mocha && npm install -g http-server && npm install -g node-sass && npm install -g babel && npm install -g browserify && npm install -g jshint",
-    "test": "npm run mocha && npm run jshint",
-    "mocha": "nyc --reporter=text -n src/converter.js mocha",
-    "coverage": "nyc report --reporter=text-lcov | coveralls",
-    "serve": "http-server dist",
     "build": "npm run css && npm run html && npm run js && npm run favicon",
     "css": "node-sass scss/style.scss dist/css/style.css --output-style compressed --source-map dist/css",
-    "html": "node ./html-compile.js",
+    "coverage": "nyc report --reporter=text-lcov | coveralls",
+    "deploy": "echo 'jsstringconverter.bbody.io' > ./dist/CNAME && delete dist/js/script.js.map && npm run js:prod && gh-pages -d dist",
```
Comment (bbody) [inline package.json]: delete me

File: html-compile.js [hunk 2]
```diff
@@ -1,20 +1,20 @@
 const pug = require('pug');
 const fs = require('fs');
 const variables = require('./index.json');
-const package = require('./package.json');
+const Package = require('./package.json');
```
Comment (bbody) [inline html-compile.js]: Why capital?

File: html-compile.js [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
 
-variables.package = package;
+variables.package = Package;
 
-const compiledFunction = pug.compileFile('index.pug');
+// const compiledFunction = pug.compileFile('index.pug');
```
Comment (bbody) [inline html-compile.js]: Why comment me out?

File: package.json (hunk 1 above)
Comment (bbody) [reply package.json]: deploy task

File: src/script.js [hunk 4]
```diff
@@ -2,40 +2,46 @@ const stringConverter = require('./converter');
...
+  const text = document.getElementById('text').value;
 
   // Get the variable name, set default as "text"
-  const variableName = document.getElementById("variable-name").value || "";
+  const variableName = document.getElementById('variable-name').value || '';
 
   // Get the type of string wanted to be output
-  const stringType = document.getElementById("string-type").value || stringConverter.ECMA5_DOUBLE;
+  const stringType = document.getElementById('string-type').value ||
+  stringConverter.ECMA5_DOUBLE;
```
Comment (bbody) [inline src/script.js]: Should be indented

File: package.json (hunk 1 above)
Comment (bbody) [reply package.json]: Can delete the jslint task and dependencies if you want

File: html-compile.js [hunk 5]
```diff
@@ -1,20 +1,20 @@
 const pug = require('pug');
 const fs = require('fs');
 const variables = require('./index.json');
-const package = require('./package.json');
+const pkg = require('./package.json');
 
-variables.package = package;
+variables.package = pkg;
```
Comment (bbody) [inline html-compile.js]:
```suggestion
variables.package = package;
```

File: html-compile.js [hunk 6]
```diff
@@ -1,20 +1,20 @@
 const pug = require('pug');
 const fs = require('fs');
 const variables = require('./index.json');
-const package = require('./package.json');
+const pkg = require('./package.json');
```
Comment (bbody) [inline html-compile.js]:
```suggestion
const package = require('./package.json');
```

Comment (obelic2) [reply html-compile.js]: The linter fails on "package" because it's a reserved word. Hence writing "pkg"

**Pull request merged**